Library/Homebrew/rubocops/conflicts_cop.rb

```diff
@@ -6,10 +6,8 @@ module Cop
     module FormulaAudit
       # This cop audits versioned Formulae for `conflicts_with`
       class Conflicts < FormulaCop
-        MSG = <<~EOS.freeze
-          Versioned formulae should not use `conflicts_with`.
-          Use `keg_only :versioned_formula` instead.
-        EOS
+        MSG = "Versioned formulae should not use `conflicts_with`. " \
```

Comment:
Struggling to see how a bunch of backslashes with heavier indentation are any more readable than an `EOS` block, personally 🙈.

Reply:
It needs to be a single line for it to work with `expect_offense`, that's the reason. `EOS` was two lines.

Reply:
Does something like `.split("\n")` not work?

Reply:
No, it only supports a single line. Look at one of the specs where `expect_offense` is used. It parses the lines with `^^^^^^^^` and matches them.

Reply:
I think this is a fairly profound regression on readability, and honestly something that would deter me from wanting to spend much time with these files, but it is what it is. Mike's approved it & I stepped away, so I'll go quietly.

Reply:
@DomT4 yes this seems weird but note two things:

1. most of the code here is still using heredocs
2. the diff here is +375 −1,167

That said, I too am struggling to understand why rubocop is forcing us into this one line situation.

@reitermarkus is this something specific to 0.51 or have we had a bug all along? If it's the former, that seems like a pretty yucky regression that we should engage with upstream about. If the latter, oh well.

Reply:
@reitermarkus Is `expect_offense` not something we define ourselves in `Library/Homebrew/test/support/helper/rubocop.rb`?

Reply:
We had `expect_offense` defined ourselves before, but now RuboCop's `rspec` support includes a `expect_offense` helper method, which they use for all of their specs.

Reply:
>Does something like .split("\n") not work?

@reitermarkus I think @DomT4 meant something like
```
        MSG = <<~EOS.chomp.tr("\n", " ").freeze
          Versioned formulae should not use `conflicts_with`.
          Use `keg_only :versioned_formula` instead.
        EOS
```

or

```
        MSG = <<~EOS.tr("\n", " ").strip.freeze
          Versioned formulae should not use `conflicts_with`.
          Use `keg_only :versioned_formula` instead.
        EOS
```
or equivalent.

Reply:
That doesn't seem more readable to me.

Reply:
Personally I think some junk on the first line is better than quotes and backslashes at the end of every line but I will leave that between you and Dom.

Reply:
> I think @DomT4 meant something like

Yeah I didn't actually do testing, I was just tossing ideas off of the top of my head in terms of could we not do this literally any other way.

> That doesn't seem more readable to me.

I agree it's not brilliant but at least from my point of view there is nothing worse than backslashes for readability. For me there is an appreciable difference between:

```ruby
blah "Polly ate three crackers and two of those had butter on " \
     "except one had salted butter and the other unsalted. " \
     "Polly wasn't sure why she was documenting her lunch to " \
     "such a degree but wondered if she should join Instagram " \
     "where documenting lunch seems to be very in vogue."
```
And:
```ruby
blah <<~EOS.tr("\n", " ").strip.freeze
  Polly ate three crackers and two of those had butter on
  except one had salted butter and the other unsalted.
  Polly wasn't sure why she was documenting her lunch to
  such a degree but wondered if she should join Instagram
  where documenting lunch seems to be very in vogue.
EOS
```
The latter has a flow that isn't punched in the face by having to check for backslashes constantly. The longer the block of text the worse it becomes. And the latter also avoids that grim necessity for whitespace before the closing `"` on each line except the final one.

Reply:
It may be worth it for many lines, but not for only 2 as in this case.

Reply:
But then you're simply introducing inconsistency across a codebase 🙈.